sql/core/src/test/resources/sql-tests/inputs/thetasketch.sql

```diff
@@ -0,0 +1,175 @@
+-- Positive test cases
+-- Create a table with some testing data.
+DROP TABLE IF EXISTS t1;
+CREATE TABLE t1 USING JSON AS VALUES (0), (1), (2), (2), (2), (3), (4) as tab(col);
+
+SELECT theta_sketch_estimate(theta_sketch_agg(col)) AS result FROM t1;
+
+SELECT theta_sketch_estimate(theta_sketch_agg(col, 12))
+FROM VALUES (50), (60), (60), (60), (75), (100) tab(col);
+
+SELECT theta_sketch_estimate(theta_sketch_agg(col))
+FROM VALUES ('abc'), ('def'), ('abc'), ('ghi'), ('abc') tab(col);
+
+SELECT theta_sketch_estimate(theta_sketch_agg(col))
+FROM VALUES (ARRAY(1, 2)), (ARRAY(3, 4)), (ARRAY(1, 2)) tab(col);
+
+SELECT theta_sketch_estimate(
+  theta_union(
+    theta_sketch_agg(col1),
+    theta_sketch_agg(col2)))
+  FROM VALUES
+    (1, 4),
+    (1, 4),
+    (2, 5),
+    (2, 5),
+    (3, 6) AS tab(col1, col2);
+
+SELECT theta_sketch_estimate(theta_union_agg(sketch, 15))
+    FROM (SELECT theta_sketch_agg(col) as sketch
+            FROM VALUES (1) AS tab(col)
+          UNION ALL
+          SELECT theta_sketch_agg(col, 20) as sketch
+            FROM VALUES (1) AS tab(col));
+
+-- Test theta_intersection
+SELECT theta_sketch_estimate(
+  theta_intersection(
+    theta_sketch_agg(col1),
+    theta_sketch_agg(col2)))
+  FROM VALUES
+    (1, 1),
+    (2, 1),
+    (3, 2),
+    (3, 3) AS tab(col1, col2);
+
+-- Test theta_difference  
+SELECT theta_sketch_estimate(
+  theta_difference(
+    theta_sketch_agg(col1),
+    theta_sketch_agg(col2)))
+  FROM VALUES
+    (1, 4),
+    (2, 4),
+    (3, 5),
+    (4, 5) AS tab(col1, col2);
+
+-- Test theta_intersection_agg
+SELECT theta_sketch_estimate(theta_intersection_agg(sketch, 15))
+    FROM (SELECT theta_sketch_agg(col) as sketch
+            FROM VALUES (1), (2) AS tab(col)
+          UNION ALL
+          SELECT theta_sketch_agg(col) as sketch
+            FROM VALUES (1), (3) AS tab(col)
+          UNION ALL  
+          SELECT theta_sketch_agg(col) as sketch
+            FROM VALUES (1), (4) AS tab(col));
+
+-- Test with explicit lgNomEntries parameters
+SELECT theta_sketch_estimate(theta_union(
+    theta_sketch_agg(col1, 12),
+    theta_sketch_agg(col2, 12), 15))
+  FROM VALUES
+    (1, 4),
+    (2, 5),
+    (3, 6) AS tab(col1, col2);
+
+SELECT theta_sketch_estimate(theta_intersection(
+    theta_sketch_agg(col1, 12),
+    theta_sketch_agg(col2, 12), 15))
+  FROM VALUES
+    (1, 1),
+    (2, 2),
+    (3, 4) AS tab(col1, col2);
+
+SELECT theta_sketch_estimate(theta_difference(
+    theta_sketch_agg(col1, 12),
+    theta_sketch_agg(col2, 12), 15))
+  FROM VALUES
+    (1, 4),
+    (2, 4),
+    (3, 5) AS tab(col1, col2);
+
+-- Negative test cases
```

Comment:
Would you mind to add a brief comment before each negative test case writing down what the erroneous input(s) are (for example, in the following query, the `lgNomEntries` value is out of range).

Reply:
yep will add more detail to the sql tests